packages/core/src/new-api/types/module.ts

```diff
@@ -31,16 +32,16 @@ export type ArgumentType =
  * @beta
  */
 export enum FutureType {
-  NAMED_CONTRACT_DEPLOYMENT,
-  ARTIFACT_CONTRACT_DEPLOYMENT,
-  NAMED_LIBRARY_DEPLOYMENT,
-  ARTIFACT_LIBRARY_DEPLOYMENT,
-  NAMED_CONTRACT_CALL,
-  NAMED_STATIC_CALL,
-  NAMED_CONTRACT_AT,
-  ARTIFACT_CONTRACT_AT,
-  READ_EVENT_ARGUMENT,
-  SEND_DATA,
+  NAMED_CONTRACT_DEPLOYMENT = "NAMED_CONTRACT_DEPLOYMENT",
```

Comment:
I replaced enums to use strings, as the type guards were getting confused between `RuntimeValueType` and `FutureType`. 

I added a test that ensures that values have to be a valid key.

Reply:
For debugging purposes, I prefer string enums anyway. But then no one ever accused me of being memory efficient.